clang/include/clang/AST/ExprCXX.h

```diff
@@ -4902,6 +4902,98 @@ class BuiltinBitCastExpr final
   }
 };
 
+/// Represents a __builtin_num_fields expression.
+class SYCLBuiltinNumFieldsExpr : public Expr {
+  friend class ASTStmtReader;
+
+  SourceLocation Loc;
+  int64_t NumFields;
```

Comment:
nit: Unsigned something or other?

Reply:
Also, do we need this one?  It might be just as easy to always just look it up from the SourceTy.

Reply:
I went ahead and removed it -- good call!